### ISSUE TITLE
About reproducing the zeroshot result on ScanObjectNN

### ISSUE THREAD

**Issue by ImageProcessing01** [tyhuang0428/CLIP2Point#2]
Dear sir,

I find the zeroshot result produced by the released code is very low: 0.089.

I download the "We provide the pre-trained checkpoint [best_eval.pt] (https://drive.google.com/file/d/1ZAnIANNMqRRRmaVtk8Kp93s_NkGU51zv/view?usp=sharing) " and use the commend: 
python zeroshot.py --ckpt best_eval.pth. 

So whether i have done the correct operation?

Thanks for your attention.

**Comment by tyhuang0428** [comment]
I haven't provided the corresponding pipeline for ScanObjectNN (maybe later?). So when you use the command: python zeroshot.py --ckpt best_eval.pth, you mean you are evaluating the zero-shot classification on ModelNet40? If that, could you please provide more information, e.g., the dataset you use, the environment information, and so on.

**Comment by ImageProcessing01** [comment]
> 
sorry for my careless. 

I use _best_eval.pth_ as the checkpoint and visual encoder.

Then I download the ModelNet40_Ply, and test the zeroshot result on it, and get the accuracy: 0.006482982171799027. 

Then I use ScanObjectNN dataset to test the zeroshot accuracy,  and get the accuracy : 0.089.

So if i made some mistake or miss something, or i use the _best_eval.pth_ in a wrong way.

I test the code on Linux and set a new environment according to your suggestions in the readme.

And thanks for your attention.

**Comment by yuchenlichuck** [comment]
The randomness of clip?

**Comment by kaimingkuang** [comment]
I modified the current zero-shot pipeline for ScanObjectNN. I got 8.95% accuracy on the OBJ_BG variation, which is lower than the paper number. You mentioned that there is a different pipeline for ScanObjectNN. When will that be open-sourced? Thanks!

**Comment by tyhuang0428** [comment]
> 

As our paper is still under submission, we may release the rest part and additional improvements altogether later this month. Hint: you may consider the rotation matric in [render.py](https://github.com/tyhuang0428/CLIP2Point/blob/main/render/render.py#L189) if you want to reproduce the result on ScanObjectNN, as its original orientation is different from ModelNet40. 

Sorry for the inconvenience. Hope you could understand that.

**Comment by CoisiniYrq** [comment]
> I haven't provided the corresponding pipeline for ScanObjectNN (maybe later?). So when you use the command: python zeroshot.py --ckpt best_eval.pth, you mean you are evaluating the zero-shot classification on ModelNet40? If that, could you please provide more information, e.g., the dataset you use, the environment information, and so on.

Could you provide the pipeline for ScanObjectNN now? Thank you.

**Comment by tyhuang0428** [comment]
> > I haven't provided the corresponding pipeline for ScanObjectNN (maybe later?). So when you use the command: python zeroshot.py --ckpt best_eval.pth, you mean you are evaluating the zero-shot classification on ModelNet40? If that, could you please provide more information, e.g., the dataset you use, the environment information, and so on.
> 
> Could you provide the pipeline for ScanObjectNN now? Thank you.

As ScanObjectNN's orientation is different from ModelNet40, the rotation process should be removed. To this end, you may set "rot=False" [here](https://github.com/tyhuang0428/CLIP2Point/blob/main/zeroshot.py#L40). Sorry for the confusing.

**Comment by CoisiniYrq** [comment]
> > > I haven't provided the corresponding pipeline for ScanObjectNN (maybe later?). So when you use the command: python zeroshot.py --ckpt best_eval.pth, you mean you are evaluating the zero-shot classification on ModelNet40? If that, could you please provide more information, e.g., the dataset you use, the environment information, and so on.
> > 
> > 
> > Could you provide the pipeline for ScanObjectNN now? Thank you.
> 
> As ScanObjectNN's orientation is different from ModelNet40, the rotation process should be removed. To this end, you may set "rot=False" [here](https://github.com/tyhuang0428/CLIP2Point/blob/main/zeroshot.py#L40). Sorry for the confusing.

Thanks for your reply, I tried this method, but it doesn't work,  when I set "rot=False", I only got 6.71% accuracy on the OBJ_BG dataset. What else do I need to set?

**Comment by tyhuang0428** [comment]
> > > > I haven't provided the corresponding pipeline for ScanObjectNN (maybe later?). So when you use the command: python zeroshot.py --ckpt best_eval.pth, you mean you are evaluating the zero-shot classification on ModelNet40? If that, could you please provide more information, e.g., the dataset you use, the environment information, and so on.
> > > 
> > > 
> > > Could you provide the pipeline for ScanObjectNN now? Thank you.
> > 
> > 
> > As ScanObjectNN's orientation is different from ModelNet40, the rotation process should be removed. To this end, you may set "rot=False" [here](https://github.com/tyhuang0428/CLIP2Point/blob/main/zeroshot.py#L40). Sorry for the confusing.
> 
> Thanks for your reply, I tried this method, but it doesn't work, when I set "rot=False", I only got 6.71% accuracy on the OBJ_BG dataset. What else do I need to set?

Have you changed the text prompt according to the labels of ScanObjectNN [here](https://github.com/tyhuang0428/CLIP2Point/blob/main/zeroshot.py#L17)?

**Comment by CoisiniYrq** [comment]
> > > > > I haven't provided the corresponding pipeline for ScanObjectNN (maybe later?). So when you use the command: python zeroshot.py --ckpt best_eval.pth, you mean you are evaluating the zero-shot classification on ModelNet40? If that, could you please provide more information, e.g., the dataset you use, the environment information, and so on.
> > > > 
> > > > 
> > > > Could you provide the pipeline for ScanObjectNN now? Thank you.
> > > 
> > > 
> > > As ScanObjectNN's orientation is different from ModelNet40, the rotation process should be removed. To this end, you may set "rot=False" [here](https://github.com/tyhuang0428/CLIP2Point/blob/main/zeroshot.py#L40). Sorry for the confusing.
> > 
> > 
> > Thanks for your reply, I tried this method, but it doesn't work, when I set "rot=False", I only got 6.71% accuracy on the OBJ_BG dataset. What else do I need to set?
> 
> Have you changed the text prompt according to the labels of ScanObjectNN [here](https://github.com/tyhuang0428/CLIP2Point/blob/main/zeroshot.py#L17)?

Yes, I changed the text prompt before set "rot=False".

**Comment by tyhuang0428** [comment]
> > > > > > I haven't provided the corresponding pipeline for ScanObjectNN (maybe later?). So when you use the command: python zeroshot.py --ckpt best_eval.pth, you mean you are evaluating the zero-shot classification on ModelNet40? If that, could you please provide more information, e.g., the dataset you use, the environment information, and so on.
> > > > > 
> > > > > 
> > > > > Could you provide the pipeline for ScanObjectNN now? Thank you.
> > > > 
> > > > 
> > > > As ScanObjectNN's orientation is different from ModelNet40, the rotation process should be removed. To this end, you may set "rot=False" [here](https://github.com/tyhuang0428/CLIP2Point/blob/main/zeroshot.py#L40). Sorry for the confusing.
> > > 
> > > 
> > > Thanks for your reply, I tried this method, but it doesn't work, when I set "rot=False", I only got 6.71% accuracy on the OBJ_BG dataset. What else do I need to set?
> > 
> > 
> > Have you changed the text prompt according to the labels of ScanObjectNN [here](https://github.com/tyhuang0428/CLIP2Point/blob/main/zeroshot.py#L17)?
> 
> Yes, I changed the text prompt before set "rot=False".

I have updated zeroshot.py to support zero-shot classification on ScanObjectNN. Please check the updated [code](). Feel free to ask if you have further questions.

**Comment by CoisiniYrq** [comment]
> > > > > > > I haven't provided the corresponding pipeline for ScanObjectNN (maybe later?). So when you use the command: python zeroshot.py --ckpt best_eval.pth, you mean you are evaluating the zero-shot classification on ModelNet40? If that, could you please provide more information, e.g., the dataset you use, the environment information, and so on.
> > > > > > 
> > > > > > 
> > > > > > Could you provide the pipeline for ScanObjectNN now? Thank you.
> > > > > 
> > > > > 
> > > > > As ScanObjectNN's orientation is different from ModelNet40, the rotation process should be removed. To this end, you may set "rot=False" [here](https://github.com/tyhuang0428/CLIP2Point/blob/main/zeroshot.py#L40). Sorry for the confusing.
> > > > 
> > > > 
> > > > Thanks for your reply, I tried this method, but it doesn't work, when I set "rot=False", I only got 6.71% accuracy on the OBJ_BG dataset. What else do I need to set?
> > > 
> > > 
> > > Have you changed the text prompt according to the labels of ScanObjectNN [here](https://github.com/tyhuang0428/CLIP2Point/blob/main/zeroshot.py#L17)?
> > 
> > 
> > Yes, I changed the text prompt before set "rot=False".
> 
> I have updated zeroshot.py to support zero-shot classification on ScanObjectNN. Please check the updated code. Feel free to ask if you have further questions.

  Thank you so much!
